### PR TITLE
[fix]: [CDS-53326]: Error message issue fixed

### PR DESCRIPTION
**Pull request by vtxorxwitty**
## Describe your changes
Error message improved when github connector is provided as runtime when versions API is called.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-ALL: `trigger tiAll`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/43972)

### CODE REVIEW

**Review by naman-goenka — APPROVED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43972)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Review by shivamnegi94 — COMMENTED** [review]
Please add UT cases

File: 120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/githubpackages/GithubPackagesArtifactResource.java [hunk 1]
```diff
@@ -75,7 +73,7 @@ public ResponseDTO<GithubPackagesResponseDTO> getPackages(@QueryParam("connector
       @NotNull @QueryParam("packageType") String packageType, @QueryParam("org") String org,
       @BeanParam GitEntityFindInfoDTO gitEntityBasicInfo) {
     if (StringUtils.isBlank(gitConnectorIdentifier)) {
-      throw new InvalidRequestException(emptyConnectorMessage);
+      throw new InvalidRequestException("Connector reference cannot be empty");
```
Comment (shivamnegi94) [inline 120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/githubpackages/GithubPackagesArtifactResource.java:76]: Error message is being same in all places, instead use static variable for Error message

**Review by shivamnegi94 — CHANGES_REQUESTED** [review]

File: 120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/githubpackages/GithubPackagesArtifactResource.java (hunk 1 above)
Comment (shivamnegi94) [reply 120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/githubpackages/GithubPackagesArtifactResource.java:76]: Please add UT

Comment (vtxorxwitty) [reply 120-ng-manager/src/main/java/io/harness/ng/core/artifacts/resources/githubpackages/GithubPackagesArtifactResource.java:76]: Initially we used a variable but that error message was not getting populated correctly. Regarding the UT, @shivamnegi94 we are not changing any logical thing over here. It's just the implementation that we are changing. And we basically don't have a Test file for the ArtifactResource Layer..

**Review by shivamnegi94 — APPROVED** [review]

**Review by akhilpandey1 — APPROVED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43972)* status: 1. unresolved discussions.

**Review by shivamnegi94 — COMMENTED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43972)* status: 1. unresolved discussions.

**Review by shivamnegi94 — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43972#-:-NOJhhm51Bj0yQUsbj77:bnfp4nl)* status: 1. unresolved discussions.

**Review by akhilpandey1 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43972)* status: 1. unresolved discussions.

**Review by shivamnegi94 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43972)* status: :shipit: complete! ✓

**Pull request merged**